openapi.yaml

```diff
@@ -960,3 +997,109 @@ components:
         approvingTeamDetails:
           type: string
           example: "DevRel"
+    SchemaRequest:
+      title: SchemaRequest
+      required:
+        - schemafull
+      type: object
+      properties:
+        req_no:
+          title: req_no
+          type: integer
+          format: int32
+        topicname:
+          title: Topic name
+          description: Kafka Topic name
+          type: string
+          example: "testtopic"
+        environment:
+          title: environment
+          description: Id of the environment
+          type: string
+          example: "3"
+        environmentName:
+          title: environmentName
+          description: Name of the environment
+          type: string
+          example: "DEV"
+        schemaversion:
+          title: schemaversion
+          description: SchemaRequest version
+          type: string
+          example: "1.0"
+        teamname:
+          title: Team name
+          description: Topic owner team name
+          type: string
+          example: "Infra"
+        teamId:
+          title: Team ID
+          description: Team identifier
+          type: integer
+          format: int32
+          example: 1010
+        appname:
+          title: App name
+          example: "App"
+          type: string
+        schemafull:
+          title: schemafull
+          maxLength: 2147483647
+          minLength: 8
+          type: string
+          example: "{\n  \"doc\": \"exampleTopic\",\n  \"fields\": [\n    {\n      \"default\": \"123456\",\n       \"doc\": \"test\",\n      \"name\": \"test\",\n      \"namespace\": \"test\",\n      \"type\": \"string\"\n }\n  ],\n  \"name\": \"exampleTopic\",\n  \"namespace\": \"exampleTopic\",\n  \"type\": \"record\"\n}"
+        username:
+          title: username
+          description: Username
+          example: jon.snow@klaw-project.io
+          type: string
+        requesttime:
+          title: Request time
+          type: string
+          format: date-time
+          example: 2018-11-13T20:20:39+00:00
+        requesttimestring:
+          title: Request time string representation
+          type: string
+          example: "28-Dec-2022 14:54:57"
+        topicstatus:
```

Comment:
I think this one is an enum, it has been documented in the  `AclRequest` schema.

Reply:
I could only find `topicstatus` as a string 🤔  anI would define a enum in our domain/types, because I'm pretty sure that this type matches what we get from the API `getRequestTypeStatuses`, but I have to check with Murali first if that's true ->`getRequestTypeStatuses` describes it's response as an array of strings.

Reply:
Yes sorry, I meant that it is defined as a string in the models, but in practice it is an enum, as in "can only be a string from a subset of possible strings". I suspect it might be the same as `aclstatus`, but of course that may be hasty until we have confirmation.

```yaml
        aclstatus:
          type: string
          enum: ["created", "approved", "denied", "deleted"]
          example: "created"
```

Reply:
@muralibasani  - am I right to assume that there is no enum for `topicstatus`? I couldn't find it in the API.

Reply:
@programmiri  It is not enum at the moment, but possible values are ["created", "approved", "declined", "deleted"] defined here https://github.com/aiven/klaw/blob/main/core/src/main/java/io/aiven/klaw/model/enums/RequestStatus.java#L4

Reply:
Awesome, thank you! Then I would leave it as it is in the OpenApi yaml (to match the current implementation) but add enums to our frontend types.